src/utils/helpers/review/getAssignedQuestions.ts

```diff
@@ -3,35 +3,34 @@ import {
   ReviewQuestionAssignment,
   TemplateElement,
 } from '../../generated/graphql'
-import { ReviewQuestion } from '../../types'
 interface GetAssignedQuestionsProps {
   reviewQuestions: ReviewQuestionAssignment[]
 }
 
+/**
+ * @function getAssignedQuestions
+ * Returns array of type ReviewQuestion with assigned questions
+ * to Reviewer from fetched in GraphQL type ReviewQuestionsAssigment
+ * @param reviewQuestions GraphQL array of assigned questions
+ */
 const getAssignedQuestions = ({ reviewQuestions }: GetAssignedQuestionsProps) => {
-  const assignedQuestions = reviewQuestions.reduce(
-    (validQuestionAssignments: ReviewQuestion[], questionAssignment) => {
+  const assignedQuestions = reviewQuestions
+    .filter((questionAssignment) => {
+      // Remove review_response without a response link
+      const { templateElement } = questionAssignment
+      const { applicationResponses } = templateElement as TemplateElement
+      return applicationResponses.nodes.length > 0
+    })
+    .map((questionAssignment) => {
+      // Map each review_response to ReviewQuestion object
       const { templateElement } = questionAssignment
       const { code, section, applicationResponses } = templateElement as TemplateElement
-      const currentResponse =
-        applicationResponses.nodes.length > 0
-          ? (applicationResponses.nodes[0] as ApplicationResponse)
-          : undefined
-
-      // Check if assigned question is valid to include in array
-      if (!currentResponse) return validQuestionAssignments
-
-      return [
-        ...validQuestionAssignments,
-        {
-          code,
-          responseId: currentResponse.id,
-          sectionIndex: section?.index as number,
-        },
-      ]
-    },
-    []
-  )
+      return {
+        code,
+        responseId: (applicationResponses.nodes[0] as ApplicationResponse).id,
+        sectionIndex: section?.index as number,
+      }
+    })
```

Comment:
Great! I find the `.map` so much clearer to understand :)

Reply:
Yes. I think it looks better too.